### ISSUE TITLE
[Q] Correlation for categorical values in parameter importance panel

### ISSUE THREAD

**Issue by CrohnEngineer** [wandb/client#2126]
Hello,

I'm a newbie in using wandb, so I would like to first thank you for such an amazing tool!  It is really improving the management of my ML projects :)
I have a small question regarding the parameter importance panel of the sweeps.
I've read the [documentation](https://docs.wandb.ai/ref/app/features/panels/parameter-importance) about it, and got the difference between importance and correlation.
Still, I was wondering: let's suppose I have a CNN, and I want to perform a hyperparameter search to find the best architecture for my task. 
Therefore, I am sweeping also on some architectural choices like the layers' activation functions, type of padding, etc.,  with a configuration file like the following:

```
name: my_sweep
metric:
  goal: minimize
  name: val_loss
program: my_train_script.py
method: grid
early_terminate:
  type: hyperband
  s: 2
  eta: 3
  max_iter: 27
parameters:
  activation:
    values: ['ReLU', 'Tanh', 'LeakyReLU']
 pad:
    values: ['zero', 'reflection']
 other_architecture_params:
     values: etc...
project:
  My-project
```

I understand the purpose of the correlation panel when we have a numerical value, like for instance the learning rate of our optimizer, but how should I interpret it when it comes to categorical values like the choice of an activation function, or of one optimizer over another?
A positive high correlation implies that the specific choice of that parameter (i.e. activation = ReLU) correlates with a high value of my metric (i.e., high value of validation loss)?

Thank you in advance for your time :)

**Comment by charlesfrye** [comment]
Hey @CrohnEngineer! Sorry for the delay -- I initially thought this question had a much more complex answer.

For categorical variables, we do [dummy coding](https://en.wikipedia.org/wiki/Dummy_variable_(statistics)), so that effectively each distinct possible value is treated independently. The correlation then tells you whether and in which direction choosing that value for that hyperparameter might move the metric.

For example, the `activation` hyperparameter is swept over in this chart, and dummy variables with names `activation.value_{name}` are reported in the importance chart. Here, choosing no activation (i.e. using a linear network) has a strong negative impact on performance, while choosing a sigmoid activation has a small positive impact on performance.

![image](https://user-images.githubusercontent.com/10442975/117706137-c93c4580-b181-11eb-9390-fccc273a57a9.png)

See [this report](https://wandb.ai/charlesfrye/mnist-sweeps/reports/Second-Pass-A-Finer-Sweep--VmlldzoxNjQ4Njc) for that plot in context. Note that the size and sign of changes in performance are relative to the average performance across all values of the hyperparameter.

**Comment by CrohnEngineer** [comment]
Hey @charlesfrye ,

thank you so much for your explanation!
I have just another silly question.

> The correlation then tells you whether and in which direction choosing that value for that hyperparameter might move the metric.

> Here, choosing no activation (i.e. using a linear network) has a strong negative impact on performance, while choosing a sigmoid activation has a small positive impact on performance.

In this case, strong negative impact means that using no activation _decreases_ the val-full/accuracy metric?
I want to make sure that the correlation is strictly numerical, so negative correlation means a numerical decrease in the metric considered, 
I'm asking since the sweep configuration allows to indicate a goal in the optimization, so I was wondering wheter the positivity/negativity of the correlation was linked by wandb smartly somehow to the optimization goal itself (i.e., for a loss function to minimize, negative correlation => worsen the optimization goal => your loss increases).
Hope I made myself clear!
Thank you again :)

Edoardo
